packages/loader/container-loader/src/test/connectionManager.spec.ts

```diff
@@ -142,4 +142,32 @@ describe("connectionManager", () => {
         assert(!oldConnection.disposed, "connection shouldn't be disposed since mock closeHandler doesn't do it - don't expect it here after fatal nack");
         assert(!mockLogger.matchEvents([{ eventName: "reconnectingDespiteFatalError" }]), "Should not see reconnectingDespiteFatalError event after fatal nack");
     });
+
+    it("should reconnect as read with leave op", async () => {
+        const connectionManager = new ConnectionManager(
+            () => mockDocumentService,
+            client as IClient,
+            true /* reconnectAllowed */,
+            mockLogger,
+            props,
+        );
+        connectionManager.connect();
+        await waitForConnection();
+        assert.strictEqual(connectionManager.connectionMode, "write", "connection mode should be write");
+
+        const message = {
```

Comment:
you should type this variable
```suggestion
        const message: ISequencedDocumentMessage = {
```